// @maxmind/minfraud-api-node/dist/src/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = exports.Processor = exports.DeliverySpeed = exports.EventType = void 0;
var EventType;
(function (EventType) {
    EventType["AccountCreation"] = "account_creation";
    EventType["AccountLogin"] = "account_login";
    EventType["EmailChange"] = "email_change";
    EventType["PasswordReset"] = "password_reset";
    EventType["PayoutChange"] = "payout_change";
    EventType["Purchase"] = "purchase";
    EventType["RecurringPurchase"] = "recurring_purchase";
    EventType["Referral"] = "referral";
    EventType["Survey"] = "survey";
})(EventType = exports.EventType || (exports.EventType = {}));
var DeliverySpeed;
(function (DeliverySpeed) {
    DeliverySpeed["SameDay"] = "same_day";
    DeliverySpeed["Overnight"] = "overnight";
    DeliverySpeed["Expedited"] = "expedited";
    DeliverySpeed["Standard"] = "standard";
})(DeliverySpeed = exports.DeliverySpeed || (exports.DeliverySpeed = {}));
var Processor;
(function (Processor) {
    Processor["Adyen"] = "adyen";
    Processor["Affirm"] = "affirm";
    Processor["Afterpay"] = "afterpay";
    Processor["Altapay"] = "altapay";
    Processor["AmazonPayments"] = "amazon_payments";
    Processor["AmericanExpressPaymentGateway"] = "american_express_payment_gateway";
    Processor["ApplePay"] = "apple_pay";
    Processor["ApsPayments"] = "aps_payments";
    Processor["Authorizenet"] = "authorizenet";
    Processor["Balanced"] = "balanced";
    Processor["Beanstream"] = "beanstream";
    Processor["Bluepay"] = "bluepay";
    Processor["Bluesnap"] = "bluesnap";
    Processor["Bpoint"] = "bpoint";
    Processor["Braintree"] = "braintree";
    Processor["Cardknox"] = "cardknox";
    Processor["Cardpay"] = "cardpay";
    Processor["Cashfree"] = "cashfree";
    Processor["Ccavenue"] = "ccavenue";
    Processor["Ccnow"] = "ccnow";
    Processor["Cetelem"] = "cetelem";
    Processor["ChasePaymentech"] = "chase_paymentech";
    Processor["CheckoutCom"] = "checkout_com";
    Processor["Cielo"] = "cielo";
    Processor["Collector"] = "collector";
    Processor["Commdoo"] = "commdoo";
    Processor["Compropago"] = "compropago";
    Processor["ConceptPayments"] = "concept_payments";
    Processor["Conekta"] = "conekta";
    Processor["Creditguard"] = "creditguard";
    Processor["Credorax"] = "credorax";
    Processor["CtPayments"] = "ct_payments";
    Processor["Cuentadigital"] = "cuentadigital";
    Processor["Curopayments"] = "curopayments";
    Processor["Cybersource"] = "cybersource";
    Processor["Dalenys"] = "dalenys";
    Processor["Dalpay"] = "dalpay";
    Processor["Datacash"] = "datacash";
    Processor["Dibs"] = "dibs";
    Processor["DigitalRiver"] = "digital_river";
    Processor["Dlocal"] = "dlocal";
    Processor["Dotpay"] = "dotpay";
    Processor["Ebs"] = "ebs";
    Processor["Ecomm365"] = "ecomm365";
    Processor["Ecommpay"] = "ecommpay";
    Processor["Elavon"] = "elavon";
    Processor["Emerchantpay"] = "emerchantpay";
    Processor["Epay"] = "epay";
    Processor["EprocessingNetwork"] = "eprocessing_network";
    Processor["Epx"] = "epx";
    Processor["Eway"] = "eway";
    Processor["Exact"] = "exact";
    Processor["FirstAtlanticCommerce"] = "first_atlantic_commerce";
    Processor["FirstData"] = "first_data";
    Processor["G2aPay"] = "g2a_pay";
    Processor["GlobalPayments"] = "global_payments";
    Processor["Gocardless"] = "gocardless";
    Processor["Heartland"] = "heartland";
    Processor["Hipay"] = "hipay";
    Processor["Ingenico"] = "ingenico";
    Processor["Interac"] = "interac";
    Processor["Internetsecure"] = "internetsecure";
    Processor["IntuitQuickbooksPayments"] = "intuit_quickbooks_payments";
    Processor["Iugu"] = "iugu";
    Processor["Klarna"] = "klarna";
    Processor["Komoju"] = "komoju";
    Processor["LemonWay"] = "lemon_way";
    Processor["MastercardPaymentGateway"] = "mastercard_payment_gateway";
    Processor["Mercadopago"] = "mercadopago";
    Processor["Mercanet"] = "mercanet";
    Processor["MerchantEsolutions"] = "merchant_esolutions";
    Processor["Mirjeh"] = "mirjeh";
    Processor["Mollie"] = "mollie";
    Processor["MonerisSolutions"] = "moneris_solutions";
    Processor["Nmi"] = "nmi";
    Processor["Oceanpayment"] = "oceanpayment";
    Processor["Oney"] = "oney";
    Processor["Onpay"] = "onpay";
    Processor["Openpaymx"] = "openpaymx";
    Processor["OptimalPayments"] = "optimal_payments";
    Processor["Orangepay"] = "orangepay";
    Processor["Other"] = "other";
    Processor["PacnetServices"] = "pacnet_services";
    Processor["Payeezy"] = "payeezy";
    Processor["Payfast"] = "payfast";
    Processor["Paygate"] = "paygate";
    Processor["Paylike"] = "paylike";
    Processor["PaymentExpress"] = "payment_express";
    Processor["Paymentwall"] = "paymentwall";
    Processor["Payone"] = "payone";
    Processor["Paypal"] = "paypal";
    Processor["Payplus"] = "payplus";
    Processor["Paysafecard"] = "paysafecard";
    Processor["Paystation"] = "paystation";
    Processor["Paytm"] = "paytm";
    Processor["Paytrace"] = "paytrace";
    Processor["Paytrail"] = "paytrail";
    Processor["Payture"] = "payture";
    Processor["Payulatam"] = "payulatam";
    Processor["Payu"] = "payu";
    Processor["Payway"] = "payway";
    Processor["Payza"] = "payza";
    Processor["Pinpayments"] = "pinpayments";
    Processor["Posconnect"] = "posconnect";
    Processor["PrincetonPaymentSolutions"] = "princeton_payment_solutions";
    Processor["Psigate"] = "psigate";
    Processor["Qiwi"] = "qiwi";
    Processor["Quickpay"] = "quickpay";
    Processor["Raberil"] = "raberil";
    Processor["Razorpay"] = "razorpay";
    Processor["Rede"] = "rede";
    Processor["Redpagos"] = "redpagos";
    Processor["Rewardspay"] = "rewardspay";
    Processor["Safecharge"] = "safecharge";
    Processor["Sagepay"] = "sagepay";
    Processor["Securetrading"] = "securetrading";
    Processor["SimplifyCommerce"] = "simplify_commerce";
    Processor["Skrill"] = "skrill";
    Processor["Smartcoin"] = "smartcoin";
    Processor["Smartdebit"] = "smartdebit";
    Processor["SolidtrustPay"] = "solidtrust_pay";
    Processor["SpsDecidir"] = "sps_decidir";
    Processor["Stripe"] = "stripe";
    Processor["Synapsefi"] = "synapsefi";
    Processor["Systempay"] = "systempay";
    Processor["Telerecargas"] = "telerecargas";
    Processor["Towah"] = "towah";
    Processor["TransactPro"] = "transact_pro";
    Processor["Tsys"] = "tsys";
    Processor["UsaEpay"] = "usa_epay";
    Processor["Vantiv"] = "vantiv";
    Processor["Verepay"] = "verepay";
    Processor["Vericheck"] = "vericheck";
    Processor["Vindicia"] = "vindicia";
    Processor["VirtualCardServices"] = "virtual_card_services";
    Processor["Vme"] = "vme";
    Processor["Vpos"] = "vpos";
    Processor["Wirecard"] = "wirecard";
    Processor["Worldpay"] = "worldpay";
})(Processor = exports.Processor || (exports.Processor = {}));
var Tag;
(function (Tag) {
    Tag["CHARGEBACK"] = "chargeback";
    Tag["NOT_FRAUD"] = "not_fraud";
    Tag["SPAM_OR_ABUSE"] = "spam_or_abuse";
    Tag["SUSPECTED_FRAUD"] = "suspected_fraud";
})(Tag = exports.Tag || (exports.Tag = {}));
