// @maxmind/minfraud-api-node/dist/src/webServiceClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const https = require("https");
const package_json_1 = require("../package.json");
const models = require("./response/models");
class WebServiceClient {
    constructor(accountID, licenseKey, timeout = 3000) {
        this.accountID = accountID;
        this.licenseKey = licenseKey;
        this.timeout = timeout;
        this.host = 'minfraud.maxmind.com';
    }
    factors(transaction) {
        return this.responseFor('factors', transaction.toString(), models.Factors);
    }
    insights(transaction) {
        return this.responseFor('insights', transaction.toString(), models.Insights);
    }
    score(transaction) {
        return this.responseFor('score', transaction.toString(), models.Score);
    }
    reportTransaction(report) {
        return this.responseFor('transactions/report', report.toString());
    }
    responseFor(path, postData, modelClass) {
        const parsedPath = `/minfraud/v2.0/${path}`;
        const url = `https://${this.host}${parsedPath}`;
        const options = {
            auth: `${this.accountID}:${this.licenseKey}`,
            headers: {
                Accept: 'application/json',
                'Content-Length': Buffer.byteLength(postData),
                'Content-Type': 'application/json',
                'User-Agent': `minfraud-api-node/${package_json_1.version}`,
            },
            host: this.host,
            method: 'POST',
            path: parsedPath,
            timeout: this.timeout,
        };
        return new Promise((resolve, reject) => {
            const req = https.request(options, (response) => {
                let data = '';
                response.on('data', (chunk) => {
                    data += chunk;
                });
                response.on('end', () => {
                    if (response.statusCode && response.statusCode === 204) {
                        return resolve();
                    }
                    try {
                        data = JSON.parse(data);
                    }
                    catch (_a) {
                        return reject(this.handleError({}, response, url));
                    }
                    if (response.statusCode && response.statusCode !== 200) {
                        return reject(this.handleError(data, response, url));
                    }
                    return resolve(new modelClass(data));
                });
            });
            req.on('error', (err) => {
                return reject({
                    code: err.code,
                    error: err.message,
                    url,
                });
            });
            req.write(postData);
            req.end();
        });
    }
    handleError(data, response, url) {
        if (response.statusCode &&
            response.statusCode >= 500 &&
            response.statusCode < 600) {
            return {
                code: 'SERVER_ERROR',
                error: `Received a server error with HTTP status code: ${response.statusCode}`,
                url,
            };
        }
        if (response.statusCode &&
            (response.statusCode < 400 || response.statusCode >= 600)) {
            return {
                code: 'HTTP_STATUS_CODE_ERROR',
                error: `Received an unexpected HTTP status code: ${response.statusCode}`,
                url,
            };
        }
        if (!data.code || !data.error) {
            return {
                code: 'INVALID_RESPONSE_BODY',
                error: 'Received an invalid or unparseable response body',
                url,
            };
        }
        return Object.assign(Object.assign({}, data), { url });
    }
}
exports.default = WebServiceClient;
